rust/rsmgp-sys/src/memgraph/mod.rs

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2016-2021 Memgraph Ltd. [https://memgraph.com]
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ffi::CStr;
+use std::rc::Rc;
+
+use crate::mgp::*;
+use crate::result::*;
+use crate::rsmgp::*;
+use crate::vertex::*;
+// Required here, if not present, tests linking fails.
+#[double]
+use crate::mgp::ffi;
+use mockall_double::double;
+
+#[derive(Debug)]
+struct MgpMemgraph {
+    args: *const mgp_list,
+    graph: *const mgp_graph,
+    result: *mut mgp_result,
+    memory: *mut mgp_memory,
+    module: *mut mgp_module,
+}
+
+/// Main object to interact with Memgraph instance.
+#[derive(Debug, Clone)]
+pub struct Memgraph {
+    memgraph: Rc<MgpMemgraph>,
```

Comment:
Why `Rc` is needed a for a struct that holds only pointers? Isn't that possible just to copy `MgpMemgraph`?

Reply:
After the discussion, we've decided to remove the whole `MgpValue` and just put all pointers to the Memgraph struct.